src/contracts/event_factory.js

```diff
@@ -25,9 +25,44 @@ export async function createTopic(args) {
     return;
   }
 
+  // Break eventName into string arrays with max length of 64
+  let eventNameStringArray = [];
+  let maxArrayLength = 10;
+  let arrayCapacity = 32 * 2; // 32 bytes times 2 hex character
+
+  let currentIdx = 0;
+  let lastIdx = eventName.length - 1;
+
+  while (currentIdx < lastIdx) {
+    console.log("currentIdx is " + currentIdx);
+    let subStr;
+    let endIdx;
+
+    if (currentIdx + arrayCapacity <= lastIdx) {
+      endIdx = currentIdx + arrayCapacity;
+      subStr = eventName.slice(currentIdx, endIdx);
+    } else {
+      endIdx = lastIdx;
+      subStr = eventName.slice(currentIdx);
+    }
+
+    // _.padEnd(subStr, arrayCapacity, '0');
+
+    currentIdx = endIdx;
+
+    // Push hex string to name array.
+    // Break the loop if max array length is reached
+    if (eventNameStringArray.length < maxArrayLength) {
+      eventNameStringArray.push('0x' + subStr);
+    } else {
+      console.log("Name array reach max length " + maxArrayLength);
+      break;
+    }
+  }
+
```

Comment:
FYI, ^ this can be simplified by using regex. I was testing with it before in the API and worked well.
https://stackoverflow.com/a/7033662/2917742

I suggest to split it to the array, and then drop any items past index 9.